tests/gis_tests/geos_tests/test_geos.py

```diff
@@ -1434,3 +1434,11 @@ def test_empty_point(self):
         self.assertEqual(p.transform(2774, clone=True), Point(srid=2774))
         p.transform(2774)
         self.assertEqual(p, Point(srid=2774))
+
+    def test_linestring_iter(self):
```

Comment:
This test doesn't catch the case when 
```python
class LineString(...):
    ...
    def __iter__(self):
        yield from self._cs
```

Reply:
Ahh, yes thanks :+1: Now it crashes in both cases.